jax/_src/numpy/lax_numpy.py

```diff
@@ -2669,9 +2669,9 @@ def arange(start, stop=None, step=None, dtype=None):
   require = partial(core.concrete_or_error, _np_asarray)
   msg = "It arose in jax.numpy.arange argument `{}`.".format
   if stop is None and step is None:
-    start = require(start, msg("stop"))
+    start = require(start, msg("start"))
     dtype = dtype or _dtype(start)
-    return lax.iota(dtype, np.ceil(start)) # avoids materializing
+    return lax.iota(dtype, int(np.ceil(start))) # avoids materializing
```

Comment:
The error string change seems wrong to me. The only mandatory argument should – while being called "start" – correspond to `stop` here; maybe we can do:
```python
stop = require(start, msg("stop"))
dtype = dtype or _dtype(stop)
return lax.iota(dtype, int(np.ceil(stop)))
```
to clarify instead? The docstring inherited from np.arange does say: `arange([start,] stop[, step,], dtype=None)`, so this seems inconsistent.

If you maintain this change, it seems you also need to change `tests/testArangeConcretizationError` in `lax_numpy_test.py`

Reply:
These are good points. I have changed back the error string, and renamed `start` to `stop` as you suggested.